lms/static/js/Markdown.Editor.js

```diff
@@ -1040,107 +1057,144 @@
         // isCancel is false if we are going to keep the text.
         var close = function (isCancel) {
             util.removeEvent(doc.body, "keydown", checkEscape);
-            var text = input.value;
+            var url = urlInput.value.trim();
+            var description = descInput.value.trim();
+
+            // reset the form to show no errors.
+            urlInput.classList.remove('has-error');
+            urlErrorMsg.style.display = 'none';
+            descInput.classList.remove('has-error');
+            descErrorMsg.style.display = 'none';
 
             if (isCancel) {
-                text = null;
+                url = null;
             }
             else {
                 // Fixes common pasting errors.
-                text = text.replace(/^http:\/\/(https?|ftp):\/\//, '$1://');
+                url = url.replace(/^http:\/\/(https?|ftp):\/\//, '$1://');
                 // doesn't change url if started with '/' (local)
-                if (!/^(?:https?|ftp):\/\//.test(text) && text.charAt(0) != '/') {
-                    text = 'http://' + text;
+                if (!/^(?:https?|ftp):\/\//.test(url) && url.charAt(0) !== '/') {
+                    url = 'http://' + url;
                 }
             }
+            var isValidUrl = /^((?:http|https|ftp):\/{2}|\/)[^]+$/.test(url),

```

Comment:
I'd like to see Jasmine tests for this new validation logic: verify with a handful of good URLs and a handful of invalid ones.